### PR TITLE
People - publish_after & expires

### PR DESCRIPTION
**Pull request by cadecairos**
1. add publish_after and expires to the People model
2. Add a filter to the API view that excludes people based on publish_after and expires.

This is a WIP and needs to be rebased and landed after #103

### CODE REVIEW

**Comment by gideonthomas** [comment]
Question: why does this depend on the "Features" app landing?

**Comment by cadecairos** [comment]
It's not so much a dependency as I accidentally based it on that feature branch instead of master.

**Comment by gideonthomas** [comment]
lol any change you can rebase? It'll be easier to review

File: app/networkapi/people/serializers.py [hunk 1]
```diff
@@ -30,5 +30,5 @@ class Meta:
             'quote',
             'affiliations',
             'internet_health_issues',
-            'partnership_logo'
+            'partnership_logo',
```
Comment (cadecairos) [inline app/networkapi/people/serializers.py]: don't know how this got here, but I can remove it if you want.

**Comment by cadecairos** [comment]
@gideonthomas I've rebased this so it's easier to review.

File: app/networkapi/people/serializers.py (hunk 1 above)
Comment (gideonthomas) [inline app/networkapi/people/serializers.py]: no keep it 👍

File: app/networkapi/people/views.py [hunk 2]
```diff
@@ -9,7 +12,11 @@ class PeopleListView(ListAPIView):
     A view that permits a GET to allow listing all the People
     in the database
     """
-    queryset = Person.objects.all()
+    now = timezone.now()
```
Comment (gideonthomas) [inline app/networkapi/people/views.py]: this is going to get unwieldy very soon. Can you follow this pattern instead?
https://github.com/mozilla/api.mozillascience.org/blob/master/app/scienceapi/events/models.py#L8-L21
https://github.com/mozilla/api.mozillascience.org/blob/master/app/scienceapi/events/models.py#L111

File: app/networkapi/people/models.py [hunk 3]
```diff
@@ -92,6 +93,18 @@ class Person(models.Model):
         help_text='Do you want to feature this person?',
         default=False,
     )
+    publish_after = models.DateTimeField(
+        help_text='Publish this person after this date and time (UTC)',
```
Comment (gideonthomas) [inline app/networkapi/people/models.py]: Let's reword this to `Make this person's profile visible only after this date and time (UTC)` and for expires `Hide this person's profile after this date and time (UTC)`

File: app/networkapi/people/serializers.py (hunk 1 above)
Comment (Pomax) [inline app/networkapi/people/serializers.py]: as for the why: python doesn't mind extra commas, but not having enough commas in certain listing syntax will completely change the meaning and do radically different things. trailing commas are, unfortunately, a good thing in Python

File: app/networkapi/features/views.py [hunk 4]
```diff
@@ -12,22 +9,13 @@ class FeatureListView(ListAPIView):
...
     in the database
     """
-    now = timezone.now()
-    queryset = Feature.objects.filter(
-        Q(publish_after__lt=now),
-        Q(expires__gt=now) | Q(expires__isnull=True),
-    )
+    queryset = Feature.objects.published()
     serializer_class = FeatureSerializer
-    pagination_class = None
```
Comment (gideonthomas) [inline app/networkapi/features/views.py]: why this change?

File: app/networkapi/people/views.py [hunk 5]
```diff
@@ -9,15 +9,14 @@ class PeopleListView(ListAPIView):
...
-    pagination_class = None
 
 
 class PersonView(RetrieveAPIView):
     """
     A view that permits a GET to allow listing a person
     in the database
     """
-    queryset = Person.objects.all()
+    queryset = Person.objects.published
```
Comment (gideonthomas) [inline app/networkapi/people/views.py]: function

File: app/networkapi/features/views.py (hunk 4 above)
Comment (cadecairos) [inline app/networkapi/features/views.py]: I don't believe it was doing anything, so I dropped it.

File: app/networkapi/people/views.py (hunk 5 above)
Comment (cadecairos) [inline app/networkapi/people/views.py]: fixed

File: app/networkapi/people/models.py [hunk 6]
```diff
@@ -92,6 +106,20 @@ class Person(models.Model):
         help_text='Do you want to feature this person?',
         default=False,
     )
+    publish_after = models.DateTimeField(
+        help_text='Make this person\'s profile visible only after '
+                  'this date and time (UTC)',
+        default=timezone.now,
```
Comment (gideonthomas) [inline app/networkapi/people/models.py]: ok so this will not work because it is only calculated once and will be the same value for all model instances.
To make it default to the current time, just pass the `auto_now_add=True` param.

Comment (cadecairos) [inline app/networkapi/people/models.py]: I was under the impression it ran that function each time it needed it?

Comment (gideonthomas) [inline app/networkapi/people/models.py]: no because it's a static field in the model, there is a param you can pass `auto_now_add` which is basically specifying the default, but it makes the form field itself non-editable, which sucks.
I opened up a pr to fix this (the trick is to do it in the form instead)

**Comment by cadecairos** [comment]
@gideonthomas is this good to go now, or were there more changes?

File: app/networkapi/features/forms.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+from django.contrib.admin.widgets import AdminSplitDateTime
+
+from networkapi.features.models import Feature
+
+
+class FeatureAdminForm(forms.ModelForm):
+    publish_after = forms.SplitDateTimeField(
+        widget=AdminSplitDateTime,
+        initial=lambda: timezone.now(),
+        help_text='Publish date of the media',
```
Comment (gideonthomas) [inline app/networkapi/features/forms.py]: whoops looks like I missed this here. Can you update it to what's in the model?

**Comment by gideonthomas** [comment]
make that one change and then r+

File: app/networkapi/features/forms.py (hunk 7 above)
Comment (cadecairos) [inline app/networkapi/features/forms.py]: Can you clarify what you mean?

Comment (gideonthomas) [inline app/networkapi/features/forms.py]: change to 'Make this feature visible only after this date and time (UTC)'
